res/controllers/Traktor-Kontrol-S3-hid-scripts.js

```diff
@@ -0,0 +1,1681 @@
+///////////////////////////////////////////////////////////////////////////////////
+/*                                                                               */
+/* Traktor Kontrol S3 HID controller script v1.00                                */
+/* Last modification: August 2020                                                */
+/* Author: Owen Williams                                                         */
+/* https://www.mixxx.org/wiki/doku.php/native_instruments_traktor_kontrol_s3     */
+/*                                                                               */
+/* For linter:                                                                   */
+/* global HIDController, HIDDebug, HIDPacket, controller                         */
+///////////////////////////////////////////////////////////////////////////////////
+/*                                                                               */
+/* TODO:                                                                         */
+/*   * should we be lighting things inside input handlers? no because we want    */
+/*   things to light up if activated in GUI, not controller.                     */
+/*   * touch for track browse, loop control, beatjump?                           */
+/*   * jog button                                                                */
+/*   * star button                                                               */
+/*                                                                               */
+///////////////////////////////////////////////////////////////////////////////////
+
+var TraktorS3 = new function() {
+    this.controller = new HIDController();
+
+    this.debugMode = false;
+
+    // ==== Friendly User Configuration ====
```

Comment:
Put this at the very top of the file before anything else. It can be a global variable.

Reply:
done